src/cache.rs

```diff
@@ -179,16 +231,36 @@ impl Cache {
             })
             .collect();
 
-        // Try to find a platform specific path first.
+        // Look up custom page (<name>.page). If it exists, return it directly
+        let custom_page = custom_dir.join(&custom_filename);
+        if custom_page.exists() && custom_page.is_file() {
+            return Some(PageLookupResult::with_page(custom_page));
+        }
+
+        // Try to find a platform specific path next, append custom patch to it.
         if let Some(pf) = self.get_platform_dir() {
-            let pf_path = Self::find_page_for_platform(&page_filename, &cache_dir, pf, &lang_dirs);
+            let pf_path = Self::find_page_for_platform(
+                &page_filename,
+                &patch_filename,
+                &cache_dir,
+                custom_dir,
+                pf,
+                &lang_dirs,
+            );
             if pf_path.is_some() {
                 return pf_path;
             }
         }
 
         // Did not find platform specific results, fall back to "common"
-        Self::find_page_for_platform(&page_filename, &cache_dir, "common", &lang_dirs)
+        Self::find_page_for_platform(
```

Comment:
As a follow up to my last comment, I now see that the patch lookup should really not be included in that function, but rather be called directly in this function here.

Looking up patches really has nothing to do with the platform or language, so it should not be handled together. This makes it harder to see, what the changing factors are in the different calls to `Self::find_page_for_platform` (for me at least)

Reply:
Okay, so I extracted the patch lookup to a separate fn. Called that in the `find_page` function and stored it in an `Option<PathBuf>` then mapped onto the result of `find_page_for_platform`